api/Trace/Events.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Trace;
+
+interface Events extends \IteratorAggregate, \Countable
+{
+    /**
+     * Adding an event should not invalidate nor change any existing iterators.
+     * @param string $name
+     * @param Attributes|null $attributes
+     * @param string|null $timestamp
+     * @return Events Must return $this to allow setting multiple attributes at once in a chain.
+     */
+    function addEvent(string $name, ?Attributes $attributes = null, ?string $timestamp = null): Events;
```

Comment:
I think that we should not accept null for Attributes and timestamp. I think we should use an empty array and an empty string as default values.
Having both empty array/string and null will make the implementation harder because both cases should be treated.

Reply:
We can't pass an array for Attributes; it's a type mismatch. I think supporting NULL is important because I would assume that most of the time you aren't adding any attributes to the event.

Similar reasons for timestamp, although we should consider validating the passed timestamp.